File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/ExpenseItemService.java
package com.github.arhor.simple.expense.tracker.service;

import java.time.LocalDate;
import java.util.List;

import com.github.arhor.simple.expense.tracker.model.ExpenseItemDTO;
import com.github.arhor.simple.expense.tracker.util.TemporalRange;

public interface ExpenseItemService {

    List<ExpenseItemDTO> getExpenseItems(Long expenseId, TemporalRange<LocalDate> dateRange);

    ExpenseItemDTO createExpenseItem(Long expenseId, ExpenseItemDTO dto);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/ExpenseItemService.kt
package com.github.arhor.simple.expense.tracker.service

import com.github.arhor.simple.expense.tracker.model.ExpenseItemDTO
import com.github.arhor.simple.expense.tracker.util.TemporalRange
import java.time.LocalDate

interface ExpenseItemService {

    fun getExpenseItems(expenseId: Long, dateRange: TemporalRange<LocalDate>): List<ExpenseItemDTO>

    fun createExpenseItem(expenseId: Long, dto: ExpenseItemDTO): ExpenseItemDTO
}
